src/interfaces/permissions-revoke.ts

```diff
@@ -0,0 +1,86 @@
+
+import { getCurrentTimeInHighPrecision } from '../utils/time.js';
+import { validateAuthorizationIntegrity } from '../core/auth.js';
+import { DwnError, DwnErrorCode } from '../core/dwn-error.js';
+import { DwnInterfaceName, DwnMethodName, Message } from '../core/message.js';
+
+export type PermissionsRevokeOptions = {
+  messageTimestamp?: string;
+  permissionsGrantId: string;
+  authorizationSignatureInput: SignatureInput;
+};
+
+export class PermissionsRevoke extends Message<PermissionsRevokeMessage> {
+  public static async parse(message: PermissionsRevokeMessage): Promise<PermissionsRevoke> {
+    await validateAuthorizationIntegrity(message);
+
+    return new PermissionsRevoke(message);
+  }
+
+  public static async create(options: PermissionsRevokeOptions): Promise<PermissionsRevoke> {
+    const descriptor: PermissionsRevokeDescriptor = {
+      interface          : DwnInterfaceName.Permissions,
+      method             : DwnMethodName.Revoke,
+      messageTimestamp   : options.messageTimestamp ?? getCurrentTimeInHighPrecision(),
+      permissionsGrantId : options.permissionsGrantId,
+    };
+
+    const authorization = await Message.signAsAuthorization(descriptor, options.authorizationSignatureInput);
+    const message: PermissionsRevokeMessage = { descriptor, authorization };
+
+    Message.validateJsonSchema(message);
+
+    return new PermissionsRevoke(message);
+  }
+
+  public async authorize(permissionsGrantMessage: PermissionsGrantMessage): Promise<void> {
+    if (this.author !== permissionsGrantMessage.descriptor.grantedFor) {
+      // Until delegation is implemented, only the DWN owner may grant or revoke access to their DWN
+      throw new DwnError(DwnErrorCode.PermissionsRevokeUnauthorizedRevoke, 'Only the DWN owner may revoke a grant');
+    }
+  }
+
+  /**
+   * @returns oldest PermissionsRevokeMessage in the array. `undefined` if given array is empty.
+   */
+  public static async getOldestRevoke(permissionsRevokeMessages: PermissionsRevokeMessage[]): Promise<PermissionsRevokeMessage | undefined> {
+    // TODO: #406 - Deduplicate code with Message.getNewestMessage (https://github.com/TBD54566975/dwn-sdk-js/issues/406)
+    let currentOldestMessage: PermissionsRevokeMessage | undefined = undefined;
+    for (const message of permissionsRevokeMessages) {
+      if (currentOldestMessage === undefined || await PermissionsRevoke.isCreatedOlder(message, currentOldestMessage)) {
+        currentOldestMessage = message;
+      }
+    }
+
+    return currentOldestMessage;
+  }
+
+  /**
+   * Checks if first message is older than second message.
+   * @returns `true` if `a` is older than `b`; `false` otherwise
+   */
+  public static async isCreatedOlder(a: PermissionsRevokeMessage, b: PermissionsRevokeMessage): Promise<boolean> {
+    // TODO: #406 - Deduplicate code with Message.isOlder (https://github.com/TBD54566975/dwn-sdk-js/issues/406)
```

Comment:
Sanity: time to remove these methods?

(Also, I have a horrible copy & paste error in the original code, it should be `aIsOlder`, thank you if you can opportunistic rename!)